eth/eventhandler/event_handler_test.go

```diff
@@ -345,6 +358,58 @@ func TestHandleBlockEventsStream(t *testing.T) {
 		require.NoError(t, err)
 		require.Equal(t, ethcommon.HexToAddress("0x1").String(), recepientData.FeeRecipient.String())
 	})
+
+	// Receive ValidatorAdded and ValidatorRemoved in one block,
+	t.Run("test ValidatorAdded + ValidatorRemoved events handling in the same block", func(t *testing.T) {
+		// create validators cluster instance
+		callableCluster := &simcontract.CallableCluster{
+			ValidatorCount:  1,
+			NetworkFeeIndex: 1,
+			Index:           1,
+			Active:          true,
+			Balance:         big.NewInt(100_000_000),
+		}
+
+		// Call the contract methods, check no err
+		_, err = boundContract.SimcontractTransactor.RegisterValidator(
+			auth,
+			validatorShares.ValidatorPubKey,
+			[]uint64{1, 2, 3, 4},
+			encodedValidatorShares,
+			big.NewInt(100_000_000),
+			*callableCluster,
+		)
+		require.NoError(t, err)
+
+		_, err = boundContract.SimcontractTransactor.RemoveValidator(
+			auth,
+			validatorShares.ValidatorPubKey,
+			[]uint64{1, 2, 3, 4},
+			*callableCluster,
+		)
+		require.NoError(t, err)
+
+		// Execute the required txs to the contract
+		sim.Commit()
+
+		// testing util shares
+		block := <-logs
+		require.NotEmpty(t, block.Logs)
+		require.Equal(t, 2, len(block.Logs))
+		require.Equal(t, ethcommon.HexToHash("0x48a3ea0796746043948f6341d17ff8200937b99262a0b48c2663b951ed7114e5"), block.Logs[0].Topics[0])
+		require.Equal(t, ethcommon.HexToHash("0xccf4370403e5fbbde0cd3f13426479dcd8a5916b05db424b7a2c04978cf8ce6e"), block.Logs[1].Topics[0])
+
+		eventsCh := make(chan executionclient.BlockLogs)
+
+		go func() {
+			defer close(eventsCh)
+			eventsCh <- block
+		}()
+
+		lastProcessedBlock, err := eh.HandleBlockEventsStream(eventsCh, false)
+		require.Equal(t, uint64(0x9), lastProcessedBlock)
+		require.NoError(t, err)
```

Comment:
I guess here other than checking the block number we could check that we are trying to execute the correct tasks?

Reply:
You're right. Working on it